csrc/id_model/id_model.cpp

```diff
@@ -392,6 +392,27 @@ ValGraph& IdModel::buildExactGraph() {
       }
     }
 
+    // Special additional mappings for ScatterOp
+    if (auto sop = dynamic_cast<ScatterOp*>(expr)) {
```

Comment:
For my own education: the logic here maps the loop domain of scatter output to its corresponding inputs;
while logical_domain_map.cpp maps the producer/consumer's logical domain.

Where are the two mapping combined to build the correct id graph?

Reply:
I'm still not 100% confident, but I made some changes and took an approach to use `TensorDomain::initial_loop_domain_` to represent mappings between the output tensor and the index and src tensors.

Reply:
Because we do not compute any index for `out` but rather it's provided by the index tensor. We only need to do the normal indexing for the index tensor, and that's where the mapping of the loop domain of the `out` tensor is important.